RHDEVDOCS-3772 Release Notes for Pipelines 1.7 GA (#44706)
openshift/openshift-docs: modules/op-release-notes-1-7.adoc

// Module included in the following assembly:
//
// * cicd/pipelines/op-release-notes.adoc
:_content-type: REFERENCE
[id="op-release-notes-1-7_{context}"]
= Release notes for {pipelines-title} General Availability 1.7

With this update, {pipelines-title} General Availability (GA) 1.7 is available on {product-title} 4.10.

[id="new-features-1-7_{context}"]
== New features

In addition to the fixes and stability improvements, the following sections highlight what is new in {pipelines-title} 1.7.

[id="pipelines-new-features-1-7_{context}"]
=== Pipelines

* With this update, the `when` expressions in a `Task` object are scoped to guard the tasks by default. To continue guarding the `Task` object and its dependent tasks, set the `scope-when-expressions-to-task` flag to `true`. 
+
[NOTE]
====

> The `scope-when-expressions-to-task` flag will be removed in a future release. As a best practice for {pipelines-shortname}, use `when` expressions scoped to the guarded `Task` only.

Reviewer: If this flag has not been deprecated in a previous release, you might need to do now.

Reviewer: Please use the standard deprecation notice format:
https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

====
// https://github.com/tektoncd/pipeline/pull/4580

* With this update, you can use variable substitution in the `subPath` field of a workspace within a task.
// https://github.com/tektoncd/pipeline/pull/4351

> * With this update, you can reference parameters and result with a bracket notation, using both single and double quotes. Prior to this update, you could only use the dot notation. For example, the following are now equivalent:

Reviewer: ```suggestion
* With this update, you can reference parameters and results by using a bracket notation with single or double quotes. Prior to this update, you could only use the dot notation. For example, the following are now equivalent:
```

** `$(param.myparam)`, `$(param['myparam'])`, and `$(param["myparam"])`. 
+

> Using the bracket notation, you can also work with parameter names containing conflicting characters, such as `"."`. For example, `$(param['my.param'])` and `$(param["my.param"])`.

Reviewer: There's nothing wrong with the style or format of this phrase. However, I think the intention of this phrase is to communicate something more like this:
```suggestion
You can use single or double quotes to enclose parameter names that contain problematic characters, such as `"."`. For example, `$(param['my.param'])` and `$(param["my.param"])`.
```

// https://github.com/tektoncd/pipeline/pull/4268

* With this update, you can include the the `onError` parameter of a step in the task definition without enabling the `enable-api-fields` flag.
// https://github.com/tektoncd/pipeline/pull/4251

[id="triggers-new-features-1-7_{context}"]
=== Triggers

> * With this update, the `feature-flag-triggers` config map has a new field `labels-exclusion-pattern`. You can set the value of this field to a regular expression (regex) pattern. The controller filters out all labels matching the regex pattern from propagating from the event listener, to the resources created for it.

Reviewer: ```suggestion
* With this update, the `feature-flag-triggers` config map has a new field `labels-exclusion-pattern`. You can set the value of this field to a regular expression (regex) pattern. The controller filters out labels that match the regex pattern from propagating from the event listener to the resources created for <name of thing>.
```
What does "it" at the end of the previous sentence refer to?

// https://github.com/tektoncd/triggers/pull/1227

> * With this update, the `TriggerGroups` field is added to the `EventListener` specification. Using this field, you can specify a set of interceptors for execution, before a group of triggers are selected and executed. To enable this feature, set the `enable-api-fields` flag in the `feature-flags-triggers` config map to `alpha`.

Reviewer: ```suggestion
* With this update, the `TriggerGroups` field is added to the `EventListener` specification. Using this field, you can specify a set of interceptors to run before selecting and running a group of triggers. To enable this feature, set the `enable-api-fields` flag in the `feature-flags-triggers` config map to `alpha`.
```
"Execute" and "execution" have very negative associations. Use "run" instead.

// https://github.com/tektoncd/triggers/pull/1232

> * With this update, Triggers support custom runs defined by a `TriggerTemplate` template.

Reviewer: ```suggestion
* With this update, `Trigger` resources support custom runs defined by a `TriggerTemplate` template.
```
"`Trigger` resources" or "triggers."
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-object-references

Reviewer: Multiple instances.

// https://github.com/tektoncd/triggers/pull/1283/files

* With this update, Triggers support emitting Kubernetes events from an `EventListener` pod.
// https://github.com/tektoncd/triggers/pull/1222

* With this update, count metrics are available for the following objects: `ClusterInteceptor`, `EventListener`, `TriggerTemplate`, `ClusterTriggerBinding`, and `TriggerBinding`.
// https://github.com/tektoncd/triggers/pull/1305

> * With his update, the `ServicePort` specification to Kubernetes resource is added. You can use it to modify on which port the event listener service is exposed. The default port is `8080`.

Reviewer: ```suggestion
* This update adds the `ServicePort` specification to Kubernetes resource. You can use this specification to modify which port exposes the event listener service. The default port is `8080`.
```

// https://github.com/tektoncd/triggers/pull/1272

* With this update, you can use the `targetURI` field in the `EventListener` specification to send cloud events during trigger processing. To enable this feature, set the `enable-api-fields` flag in the `feature-flags-triggers` config map to `alpha`.
// https://github.com/tektoncd/triggers/pull/1259

* With this update, the `tekton-triggers-eventlistener-roles` object now has a `patch` verb, in addition to the `create` verb that already exists.
// https://github.com/tektoncd/triggers/pull/1291

* With this update, the `securityContext.runAsUser` parameter is removed from event listener deployment.
// https://github.com/tektoncd/triggers/pull/1213

[id="cli-new-features-1-7_{context}"]
=== CLI

> * With this update, the `tkn [pipeline | pipelinerun] export` command will export a pipeline or pipeline run as a YAML file. For example:

Reviewer: Avoid future tense - "will."

** Export a pipeline named `test_pipeline` in the `openshift-pipelines` namespace:
+
[source,terminal]
----
$ tkn pipeline export test_pipeline -n openshift-pipelines
----
** Export a pipeline run named `test_pipeline_run` in the `openshift-pipelines` namespace:
+
[source,terminal]
----
$ tkn pipelinerun export test_pipeline_run -n openshift-pipelines
----
// https://github.com/tektoncd/cli/pull/1398 and https://github.com/tektoncd/cli/pull/1500

> * With this update, the `--grace` option is added to the `tkn pipelinerun cancel`. Use the `--grace` option to gracefully terminate a pipeline run, instead of forced termination. To enable this feature, set the `enable-api-fields` flag in the `feature-flags` config map to `alpha`.

Reviewer: ```suggestion
* With this update, the `--grace` option is added to the `tkn pipelinerun cancel`. Use the `--grace` option to terminate a pipeline run gracefully instead of forcing the termination. To enable this feature, set the `enable-api-fields` flag in the `feature-flags` config map to `alpha`.
```

// https://github.com/tektoncd/cli/pull/1479

> * With this update, the Operator version and the Chains version is added to the output of the `tkn version` command.

Reviewer: ```suggestion
* This update adds the Operator and Chains versions to the output of the `tkn version` command.
```

+
[IMPORTANT]
====
Tekton Chains is a Technology Preview feature.
====
// https://github.com/tektoncd/cli/pull/1486 and https://github.com/tektoncd/cli/pull/1509

> * With this update, the `tkn pipelinerun describe` command displays all cancelled task runs, when you cancel a pipeline run. Before this fix, only one task run was displayed.

Reviewer: ```suggestion
* With this update, the `tkn pipelinerun describe` command displays all canceled task runs, when you cancel a pipeline run. Before this fix, only one task run was displayed.
```
Where variations exist, we use the US English spelling of words.

Reviewer: Avoid using commas before a dependent clause like this.

// https://github.com/tektoncd/cli/pull/1482

> * With this update, you can skip supplying the asking specifications for optional workspace, when you run the `tkn [t | p | ct] start` command skips with the `--skip-optional-workspace` flag. You can also skip it when running in interactive mode.

Reviewer: ```suggestion
* With this update, you can skip supplying the asking specifications for optional workspace when you run the `tkn [t | p | ct] start` command skips with the `--skip-optional-workspace` flag. You can also skip it when running in interactive mode.
```
Avoid using commas before a dependent clause like this.

// https://github.com/tektoncd/cli/pull/1465

> * With this update, you can use the `tkn chains` command to manage Tekton Chains. You can also use the `--chains-namespace` option to specify the namespace in which you want to install Tekton Chains.

Reviewer: ```suggestion
* With this update, you can use the `tkn chains` command to manage Tekton Chains. You can also use the `--chains-namespace` option to specify the namespace where you want to install Tekton Chains.
```

+
[IMPORTANT]
====
Tekton Chains is a Technology Preview feature.
====
// https://github.com/tektoncd/cli/pull/1440 and https://github.com/tektoncd/cli/pull/1522

[id="operator-new-features-1-7_{context}"]
=== Operator

* With this update, you can use the {pipelines-title} Operator to install and deploy Tekton Hub and Tekton Chains.
+
[IMPORTANT]
====
Tekton Chains and Tekton Hub are Technology Preview features.
====
// https://github.com/tektoncd/operator/pull/467, https://github.com/tektoncd/operator/pull/479, https://github.com/tektoncd/operator/pull/467, https://github.com/tektoncd/operator/pull/630, and https://github.com/tektoncd/operator/pull/630

* With this update, you can find and use Pipelines as Code (PAC) as an add-on option.
+
[IMPORTANT]
====
Pipelines as Code is a Technology Preview feature.
====
// https://github.com/tektoncd/operator/pull/550

> * With this update, you can now disable installation of community cluster tasks by setting the `communityClusterTasks` parameter to `false`. For example:

Reviewer: ```suggestion
* With this update, you can now disable the installation of community cluster tasks by setting the `communityClusterTasks` parameter to `false`. For example:
```

+
[source,yaml]
----
...
spec:
  profile: all
  targetNamespace: openshift-pipelines
  addon:
    params:
    - name: clusterTasks
      value: "true"
    - name: pipelineTemplates
      value: "true"
    - name: communityClusterTasks
      value: "false"
...
----
// https://github.com/tektoncd/operator/pull/658

> * With this update, you can disable the integration of Tekton Hub with the **Developer** perspective, by setting the `enable-devconsole-integration` flag in the `TektonConfig` custom resource to `false`. For example:

Reviewer: ```suggestion
* With this update, you can disable the integration of Tekton Hub with the **Developer** perspective by setting the `enable-devconsole-integration` flag in the `TektonConfig` custom resource to `false`. For example:
```
Dependent clause. No comma.

+
[source,yaml]
----
...
hub:
  params:
    - name: enable-devconsole-integration
      value: "true"
...
----
// https://github.com/tektoncd/operator/pull/569 

* With this update, the `operator-config.yaml` cong map enables display of the Operator version as part of the output of the `tkn version` command.
// https://github.com/tektoncd/operator/pull/563

* With this update, the version of the `argocd-task-sync-and-wait` tasks is modified to `v0.2`.
// https://github.com/tektoncd/operator/pull/642

* With this update to the `TektonConfig` CRD, the `oc get tektonconfig` command displays the OPerator version.
// https://github.com/tektoncd/operator/pull/644

* With this update, service monitor is added to the Triggers metrics.
// https://github.com/tektoncd/operator/pull/635

[id="hub-new-features-1-7_{context}"]
=== Hub

[IMPORTANT]
====

> Tekton Hub on cluster is a Technology Preview feature.

Reviewer: Do you mean...
```suggestion
Deploying Tekton Hub on a cluster is a Technology Preview feature.
```

====

Tekton Hub helps you discover, search, and share reusable tasks and pipelines for your CI/CD workflows. A public instance of Tekton Hub is available at link:https://hub.tekton.dev/[hub.tekton.dev]. 

Staring with {pipelines-title} 1.7, cluster administrators can also install and deploy a custom instance of Tekton Hub on enterprise clusters. You can curate a catalog with reusable tasks and pipelines specific to your organization.

[id="chains-new-features-1-7_{context}"]
=== Chains

[IMPORTANT]
====
Tekton Chains is a Technology Preview feature.
====

Tekton Chains is a Kubernetes Custom Resource Definition (CRD) controller. You can use it to manage the supply chain security of the tasks and pipelines created using {pipelines-title}.

> By default, Tekton Chains observes all task run executions in your {product-title} cluster. When the task runs complete, Chains takes a snapshot of the task runs. It then converts the snapshot to one or more standard payload formats, and finally signs and stores all artifacts.

Reviewer: ```suggestion
By default, Tekton Chains monitors the task runs in your {product-title} cluster. Chains takes snapshots of completed task runs, converts them to one or more standard payload formats, and signs and stores all artifacts.
```
- Executions etc.
- "observes" seems like anthropomorphism - how about monitors instead?

Tekton Chains supports the following features:

* You can sign task runs, task run results, and OCI registry images with cryptographic key types and services such as `cosign`.

* You can use attestation formats such as `in-toto`.

* You can securely store signatures and signed artifacts using OCI repository as a storage backend.

[id="pac-new-features-1-7_{context}"]
=== Pipelines as Code (PAC)

[IMPORTANT]
====
Pipelines as Code is a Technology Preview feature.
====

> With Pipelines as Code, cluster administrators and users with the required privileges can define pipeline templates as part of source code Git repositories. When triggered by source code push or a pull request for the configured Git repository, the feature runs the pipeline and reports status.

Reviewer: ```suggestion
With Pipelines as Code, cluster administrators and users with the required privileges can define pipeline templates as part of source code Git repositories. When triggered by a source code push or a pull request for the configured Git repository, the feature runs the pipeline and reports status.
```

Pipelines as Code supports the following features:

> * Pull request status. When iterating over a pull request, the status and control is executed on the platform hosting the Git repository.

Reviewer: Executed => performed or runs? Multiple instances.

Reviewer: ```suggestion
* Pull request status. When iterating over a pull request, the status and control are performed on the platform hosting the Git repository.
```
- "is" => "are" for multiple subjects.
- This phrase is still kind of odd. Does one really execute (or perform) status and control?

Author: Pull request status. When iterating over a pull request, the status and control of the pull request is **exercised** on the platform hosting the Git repository.

> * GitHub Checks API to set the status of a pipeline run, including rechecks.

Reviewer: Do you mean...
```suggestion
* GitHub checks the API to set the status of a pipeline run, including rechecks.
```

> * GitHub Pull Request and Commit events. 

Reviewer: Lower case?
```suggestion
* GitHub pull request and commit events. 
```

* Pull request actions in comments, such as `/retest`.

* Git events filtering, and a separate pipeline for each event.

> * Automatic task resolution in {pipelines-shortname} (local tasks, Tekton Hub, and remote URLs).

Reviewer: ```suggestion
* Automatic task resolution in {pipelines-shortname} for local tasks, Tekton Hub, and remote URLs.
```
Avoid using inline parenthesis like this.

* Use of GitHub blobs and objects API for retrieving configurations.

> * Access Control List (ACL) over a GitHub organization, or using a Prow style `OWNER` file.

Reviewer: over = from?

* The `tkn-pac` plugin for the `tkn` CLI tool, which you can use to manage {pac} repositories and bootstrapping.

* Support for GitHub Application, GitHub Webhook, Bitbucket Server, and Bitbucket Cloud.

[id="deprecated-features-1-7_{context}"]
== Deprecated features

// Pipelines
* Breaking change: The `disable-working-directory-overwrite` and `disable-home-env-overwrite` fields in the `TektonConfig` custom resource (CR) is removed. As a result, the `$HOME` environment variable and `workingDir` parameter are unavailable.
// https://github.com/tektoncd/pipeline/pull/4587

> * The `Conditions` custom resource definition (CRD) type will be removed in a future release. Instead, use the recommended `When` expression.

Reviewer: Please use the standard deprecation notice format:
https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

Author: The `Conditions` custom resource definition (CRD) type is deprecated and planned to be removed in a future release.

// issue # unknown; discussed in Slack.

// Triggers

> * Breaking change: Triggers validates and gives error if you provide empty `EventListener` and `TriggerBinding` specifications.

Reviewer: ```suggestion
* Breaking change: The `Triggers` resource validates <something> and generates an error if you do not specify `EventListener` and `TriggerBinding` values.
```

// https://github.com/tektoncd/triggers/pull/1277 and https://github.com/tektoncd/triggers/pull/1264

[id="known-issues-1-7_{context}"]
== Known issues

// Pipelines

> * Implicit parameter mapping incorrectly passes parameters from the top-level `Pipeline` or `PipelineRun` definitions to the `taskRef` tasks. Mapping should only occur from a top-level resource to tasks with in-line `taskSpec` specifications. This issue only affects users who have set `enable-api-fields` feature flag to `alpha`.

Reviewer: ```suggestion
* Implicit parameter mapping incorrectly passes parameters from the top-level `Pipeline` or `PipelineRun` definitions to the `taskRef` tasks. Mapping should only occur from a top-level resource to tasks with in-line `taskSpec` specifications. This issue only affects users who have set the `enable-api-fields` feature flag to `alpha`.
```

// https://github.com/tektoncd/pipeline/issues/4483

[id="fixed-issues-1-7_{context}"]
== Fixed issues

// Pipelines
* With this fix, metadata such as `labels` and `annotations` from task and pipeline definitions does not propagate to task runs and pipeline runs.
// https://github.com/tektoncd/pipeline/pull/4478

* With this fix, if the `timeouts.tasks` field or the `timeouts.finally` field is set to `0`, then the `timeouts.pipeline` is also set to `0`. 
// https://github.com/tektoncd/pipeline/pull/4539

* With this fix, the `-x` set flag is removed from scripts that do not use a shebang. The fix reduces potential data leak from script execution.
// https://github.com/tektoncd/pipeline/pull/4451

> * With this fix, any backslash character present in the usernames in Git credentials will be escaped with an additional backslash in the `.gitconfig` file.

Reviewer: ```suggestion
* With this fix, any backslash character present in the usernames in Git credentials is escaped with an additional backslash in the `.gitconfig` file.
```
Are you saying the system does this, or that the user can do it by adding a backslash?

Author: System.

// https://github.com/tektoncd/pipeline/pull/4337

// Triggers
* With this fix, the `finalizer` property of the `EventListener` object is not necessary for cleaning up logging and config maps.
// https://github.com/tektoncd/triggers/pull/1244

* With this fix, the default HTTP client associated with the event listener server is removed, and a custom HTTP client added. As a result, the timeouts have improved.
// https://github.com/tektoncd/triggers/pull/1308

> * With this fix, the Triggers cluster role now works with owner references.

Reviewer: Consider replacing all instances of "With this fix," with "With this update,"

// https://github.com/tektoncd/triggers/pull/1267

* With this fix, the race condition in the event listener does not happen when multiple interceptors return extensions.
// https://github.com/tektoncd/triggers/pull/1282

// CLI

> * With this fix, the `tkn pr delete` command does not the pipeline runs with the `ignore-running` flag.

Reviewer: Does not what?

// https://github.com/tektoncd/cli/pull/1532

// Operator

> * With this fix, the Operator pods will not continue restarting when you modify any add-on parameters.

Reviewer: ```suggestion
* With this fix, the Operator pods do not continue restarting when you modify any add-on parameters.
```

// https://github.com/tektoncd/operator/pull/631

> * With this fix, the `tkn serve` CLI pod will be scheduled on infra nodes, if not configured in the subscription and config custom resources.

Reviewer: ```suggestion
* With this fix, the `tkn serve` CLI pod is scheduled on infra nodes, if not configured in the subscription and config custom resources.
```
Avoid "will"/future tense.

Reviewer: ```suggestion
* With this update, if you do not configure the `tkn serve` CLI pod in the subscription and config custom resources, the scheduler runs it on the infra nodes.
```
- Please use the standard release note templates. If not, adopt some language from them. 
- Also, it's hard for me to understand this release note. This suggestion is my best guess at the intended meaning.
